otwcode/otwarchive
# AO3-5793 Docker support

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5793

## Purpose

This PR adds support for development/testing in Docker.

## Testing Instructions

I'll update the [wiki page](https://github.com/otwcode/otwarchive/wiki/Docker) once this PR is approved. But in essence, to test this you should:

- change the default password from `change_me` to a new secret in `docker-compose.yml` and `config/database.yml`
- run `./docker/init.sh` to build and setup docker fixtures (images, database migrations, etc.)
- run `docker-compose up web` to run the Archive locally

## References

See issue #3661 for previous discussion and contributions.

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

Tom Milligan

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

they/them
